### DISCUSSION TITLE
Verisure DoorLock (Danalock)

### DISCUSSION THREAD

**Discussion by SvenNDH** [ptz0n/homebridge-verisure#153] [General]
Hi, I have Verisure in Germany and added the Plugin to my Homebridge. But I can't use the Doorlock in HomeKit - it is not shown .

Anyone have tips for me ?

**Comment by ptz0n** [comment]
Hi and thanks for reaching out.

Have you added `doorCode` to your config?

**Reply by SvenNDH** [reply]
yes I added the doorCode to my config

 
            ],
            "alarmCode": "****",
            "doorCode": "****",
            "installations": [
                "****"
            ],
            "pollInterval": 60,
            "showAutoLockSwitch": true,
            "showDoorBellSwitch": true,
            "showLockSwitch": true,
            "showAudioSwitch": true,
            "audioOffValue": "SILENCE",
            "audioOnValue": "HIGH",
            "platform": "verisure",
            "_bridge": {
                "username": "****",
                "port": ****

**Reply by ptz0n** [reply]
I see a unknown config property in there: `showDoorBellSwitch`

Any details in your logs?

**Reply by SvenNDH** [reply]
OK i deleted the unknown thing.

Logs are ok, no errors. But in the logs only the fire detectors and the window / door contacts listed.

**Reply by ptz0n** [reply]
Strange. Do you have these locks available and controllable in your Verisure app?

**Reply by SvenNDH** [reply]
yes the door lock is available and controllable in my Verisure app. 
That's crazy

**Reply by ptz0n** [reply]
Oh, what kind of lock do you have? Yale or Danalock?

**Reply by SvenNDH** [reply]
I dint know, virisure calls it Lockguard and its Verisure branded

**Comment by ptz0n** [comment]
Yup, thats Danalock. Let me have a look to see if we can support it 👌🏻

**Comment by SvenNDH** [comment]
Thank you ☺️

**Comment by behradmoini** [comment]
did you guys ever find the solution i cant see my danalocks either 

i saw this 

const VerisureAccessory = require('./verisure');
const {
  overviewOperation,
  doorLockOperation,
  doorUnlockOperation,
  pollLockStateOperation,
  doorLockConfigOperation,
  doorLockUpdateConfigOperation,
} = require('../operations');
class DoorLock extends VerisureAccessory {
  constructor(...args) {
    super(...args);
    const { area } = this.config.device;
    this.name = VerisureAccessory.getUniqueAccessoryName(`SmartLock (${area})`);
    this.doorCode = this.platformConfig.doorCode;
    this.switchName = VerisureAccessory.getUniqueAccessoryName(`Auto-lock (${area})`);
    this.audioName = VerisureAccessory.getUniqueAccessoryName(`Audio (${area})`);
    // TODO: Could set currentLockState here.
  }
  resolveCurrentLockState({ currentLockState, motorJam }) {
    const { LockCurrentState } = this.homebridge.hap.Characteristic;
    if (motorJam) {
      return LockCurrentState.JAMMED;
    }
    return currentLockState === 'LOCKED'
      ? LockCurrentState.SECURED
      : LockCurrentState.UNSECURED;
  }

  getDoorLockState() {
    this.log('Getting current lock state.', 'debug');

    return this.installation.client(overviewOperation)
      .then((overview) => overview.installation.doorlocks
        .find((doorlock) => doorlock.device.deviceLabel === this.serialNumber))
@@ -47,8 +49,6 @@ class DoorLock extends VerisureAccessory {
  }

  getCurrentLockState(callback) {
    this.log('Getting current lock state.', 'debug');

    // TODO: Use this.cachedValue, if available?

    this.getDoorLockState().then((doorLock) => {
@@ -117,7 +117,6 @@ class DoorLock extends VerisureAccessory {

    this.installation.client(doorLockConfigOperation(this.serialNumber))
      .then(({ installation: { smartLocks: [smartLock] } }) => {
        this.log(`Got config: ${JSON.stringify(smartLock)}`, 'debug');
        this.autoLockState = DoorLock.resolveAutoLockState(smartLock.configuration);
        callback(null, this.autoLockState);
      })
@@ -145,7 +144,6 @@ class DoorLock extends VerisureAccessory {

    this.installation.client(doorLockConfigOperation(this.serialNumber))
      .then(({ installation: { smartLocks: [smartLock] } }) => {
        this.log(`Got config: ${JSON.stringify(smartLock)}`, 'debug');
        this.audioState = smartLock.configuration.volume !== this.platformConfig.audioOffValue;
        callback(null, this.audioState);
      })
@@ -159,6 +157,8 @@ class DoorLock extends VerisureAccessory {
      ? this.platformConfig.audioOnValue
      : this.platformConfig.audioOffValue;

    this.log(`Setting audio volume to: ${volume}`);

    this.installation.client(
      doorLockUpdateConfigOperation(this.serialNumber, { volume }),
    )
      .then(() => callback(null))
      .catch((error) => {
        this.log(error.message, 'debug');
        // TODO: Revert config state in this.value?
        callback({ message: error.errorMessage });
      });
  }
  getServices() {
    const { Service, Characteristic } = this.homebridge.hap;
    const services = [this.accessoryInformation];
    this.lockService = new Service.LockMechanism(this.name);
    const currentStateCharacteristic = this.lockService
      .getCharacteristic(Characteristic.LockCurrentState)
      .on('get', this.getCurrentLockState.bind(this));
    this.lockService
      .getCharacteristic(Characteristic.LockTargetState)
      .on('get', this.getTargetLockState.bind(this))
      .on('set', this.setTargetLockState.bind(this));
    services.push(this.lockService);
    if (this.platformConfig.showAutoLockSwitch) {
      this.autoLockService = new Service.Switch(this.switchName, this.switchName);
      this.autoLockService
        .getCharacteristic(Characteristic.On)
        .on('get', this.getAutoLockState.bind(this))
        .on('set', this.setAutoLockState.bind(this));
      services.push(this.autoLockService);
    }
    if (this.platformConfig.showAudioSwitch) {
      this.audioService = new Service.Switch(this.audioName, this.audioName);
      this.audioService
        .getCharacteristic(Characteristic.On)
        .on('get', this.getAudioState.bind(this))
        .on('set', this.setAudioState.bind(this));
      services.push(this.audioService);
    }
    this.pollCharacteristics.push(currentStateCharacteristic);
    return services;
  }
}
module.exports = DoorLock;

but i have no idea how to implement it 

thank you in advance
